### ISSUE TITLE
Automatically focus on first text-field in a formspec

### ISSUE THREAD

**Issue by ClobberXD** [minetest/minetest#7406]
##### Issue type

 - Feature request
 
##### Summary

Many simple formspecs just have a text-field and an OK button. The text-field (the first, if there are many) should be auto-focused when a formspec is displayed, instead of players having to manually click on them.

**Comment by paramat** [comment]
This may not be good for some formspecs though. It's annoying in inventory to not be able to exit again using the inventory key because it focusses on search which is rarely used (a mistake).

**Comment by ClobberXD** [comment]
Agreed, but Unified Inventory was just an example. My proposal isn't appropriate for UI, because the search is an _auxillary_ function. :)

There are many simple formspecs provided by mods (none of which I seem to remember, unfortunately), that just contain one text-field and one OK button. Yet the player has to manually click on the text-field for it to start capturing the key-strokes.

**Comment by paramat** [comment]
Yes useful for that situation. What i suggest is making this an option.

**Comment by ClobberXD** [comment]
I think we'd benefit from a new formspec element, which will allow mods to optionally define a custom field to be focused when displayed.

e.g.
```
fs = "field[1,1;5,1;test;Field]" ..
     "set_focus[test]"
```

**Comment by rubenwardy** [comment]
I'm fairly sure this already happens, but first selectable element rather than first field
